code/modules/overmap/ftl_shunt/core.dm

```diff
@@ -0,0 +1,551 @@
+			switch(sabotaged)
+				if(SHUNT_SABOTAGE_MINOR)
+					to_chat(user, SPAN_WARNING("It looks like it's been tampered with in some way, and the accelerator vanes seem out of place."))
+				if(SHUNT_SABOTAGE_MAJOR)
+					to_chat(user, SPAN_WARNING("Light behaves oddly around the core of [src], and it looks to have been tampered with! The vanes are definitely out of place."))
+				if(SHUNT_SABOTAGE_CRITICAL)
+					to_chat(user, SPAN_DANGER("Light bends around the core of [src] in a manner that eerily reminds you of a singularity... the vanes look completely misaligned!"))
+		else
+			to_chat(user, SPAN_WARNING("It looks like it's been tampered with, but you're not sure about to what extent."))
+
+/obj/machinery/ftl_shunt/core/attackby(var/obj/item/O, var/mob/user)
+	if(istype(O, /obj/item/stack/telecrystal))
+		var/obj/item/stack/telecrystal/TC = O
+		var/tc_input = input(user, "How many telecrystals do you want to put in?", "TC Input", 0) as num|null
+
+		if(!tc_input)
+			return FALSE
+
+		if(TC.amount < tc_input)
+			to_chat(user, SPAN_WARNING("You don't have enough telecrystals for that."))
+			return FALSE
+
+		if(tc_input < 10)
+			to_chat(user, SPAN_WARNING("This really isn't a useful amount for sabotage."))
+			return FALSE
+
+		to_chat(user, SPAN_WARNING("You begin to insert the crystals into [src]..."))
+
+		if(!do_after(user, 6 SECONDS, src))
+			return FALSE
+
+		switch(tc_input)
+			if(10 to 24)
+				sabotaged = SHUNT_SABOTAGE_MINOR
+			if(25 to 49)
+				sabotaged = SHUNT_SABOTAGE_MAJOR
+			if(50 to INFINITY)
+				sabotaged = SHUNT_SABOTAGE_CRITICAL
+
+		sabotaged_amt = tc_input
+		TC.use(tc_input)
+		to_chat(user, SPAN_DANGER("You successfully sabotage [src] by inserting the crystals!"))
+		return TRUE
+	. = ..()
+
+/obj/machinery/ftl_shunt/core/physical_attack_hand(var/mob/user)
+	if(sabotaged)
+		var/mob/living/carbon/human/h_user = null
+		if (!istype(user, /mob/living/carbon/human))
+			return
+		h_user = user //this is used specifically in the do_after because electrocute_act() is not a moblevel proc.
+		var/skill_delay = user.skill_delay_mult(SKILL_ENGINES, 0.3)
+		if(!user.skill_check(SKILL_ENGINES, SKILL_BASIC))
+			to_chat(user, SPAN_DANGER("You are nowhere near experienced enough to stick your hand into that thing."))
+			return FALSE
+		to_chat(user, SPAN_NOTICE("You reach your hand inside of [src] and slowly begin to remove re-align the accelerator vanes..."))
+		if(!do_after(user, (4 SECOND * skill_delay), src))
+			to_chat(user, SPAN_WARNING("You try to pull your hand away from the vanes, but you touch a conductor!"))
+			h_user.electrocute_act(rand(150,250), src, def_zone = user.get_active_held_item_slot())
+			return FALSE
+		var/obj/item/stack/telecrystal/TC = new
+		TC.amount = sabotaged_amt
+		TC.forceMove(get_turf(user))
+		user.put_in_hands(TC)
+		to_chat(user, SPAN_NOTICE("You realign the accelerator vanes by removing some kind of crystal, preventing what could have been a catastrophe."))
```

Comment:
Nit: I feel like `You remove \the [TC] from \the [src] and realign the accelerator vanes, preventing what could have been a catastrophe.` would flow a bit better. It would also not presume that the person doesn't know what a TC is IC.